src/redemptions/redemption.service.ts

```diff
@@ -138,6 +150,59 @@ export class RedemptionService {
     return null;
   }
 
+  async repeatedFaceCheck(
+    response: JumioTransactionRetrieveResponse,
+  ): Promise<{ status: KycStatus; failureMessage: string } | null> {
+    // can only accept if no other errors occurred AND repeated face is from previous transaction of SAME user
+    let potentialFailures: string[] = [];
+    potentialFailures = potentialFailures.concat(
+      response.capabilities.dataChecks.map((i) => i.decision.type),
+      response.capabilities.extraction.map((i) => i.decision.type),
+      response.capabilities.liveness.map((i) => i.decision.type),
+      response.capabilities.similarity.map((i) => i.decision.type),
+      response.capabilities.usability.map((i) => i.decision.type),
+    );
+    // if any other check failed, we can't pass
+    for (const potentialFailure of potentialFailures) {
+      if (potentialFailure !== 'PASSED') {
+        return null;
+      }
+    }
+    // for imageChecks, only duplicate face should pass
+    const failedImageChecks = response.capabilities.imageChecks.filter(
+      (i) => i.decision.type !== 'PASSED',
+    );
+    let duplicateFaceWorkflows: string[] = [];
+    for (const imageCheck of failedImageChecks) {
+      if (imageCheck.decision.details.label !== 'REPEATED_FACE') {
+        return null;
```

Comment:
I see what you are saying, going to refactor the code a bit based on your other suggestions and see if this comment still matters. I don't think it will :)